### DISCUSSION TITLE
Suggested update to Building an RS-485 Adapter Help Page

### DISCUSSION THREAD

**Discussion by djlevin** [CharlesGodwin/pymagnum#47] [Ideas]
Fabulous project, and VERY much needed.  Thanks so much for the work on this!

I am building my system, and after reading the help page on making up an RS-485 adapter, I have a suggested edit.

I _think_ that the easiest way to build a suitable adapter is to buy two parts from e.g. Amazon.

1) Male RJ11 to 4 PIN AV screw terminal Adapter such as: https://a.co/coa2gwv

2) RS485 to USB Adapter, cable terminated such as: https://a.co/7T4iNq3

Then I think one can just connect the red and green wires from the cable end to the middle two screws on the RJ11 adapter, and you are done.  Easy and cheap.

**Comment by CharlesGodwin** [comment]
Looks like a great idea.

**Reply by djlevin** [reply]
As it turns out, you need to connect the green and white wires to the outside two screws of the adapter, and make sure that the black and red wires do not short to each other.  If you do that, works perfectly.

**Comment by Shadow-Genesis** [comment]
I need some help setting up my adapter. I'm not sure if I got the right pins. I got a rj-11 6p6c cable. I see it is only using 4 pins in the docs. Dose it start at the pin 1(Outside) or pin 2(using the middle 4 pins)?

**Reply by CharlesGodwin** [reply]
If you're using a 6 pin setup it 2 and 5. If it's 4 pin, it's 1 and 4.

**Reply by Shadow-Genesis** [reply]
Thanks, I just tested that before seeing your reply and it did work. Sorry I should have done that first, but its now here for the next.

**Comment by CharlesGodwin** [comment]
No problem. Sometimes this stuff can be intimidating

________________________________

**Comment by kevinherzig** [comment]
Dropping a line to say that I've had terrific luck with this adapter from Amazon and it has greatly simplified the build.  

https://www.amazon.com/dp/B0DR218X7P?ref_=ppx_hzsearch_conn_dt_b_fed_asin_title_6

**Reply by CharlesGodwin** [reply]
Looks great. I have questions:
How do you deal with polarity? If the polarity of pin 1 and 4 are reversed.
Is that an RJ-11 connector or RJ45 (8 pin)

**Reply by kevinherzig** [reply]
It's an RJ-45.  You probably know this but RJ45 jacks are designed to accept RJ-11 plugs.  I didn't have to do anything with polarity, I just plugged in the RJ11 cable which was connected to a splitter behind the remote, ran magtest and it worked perfectly.
